packages/dds/merge-tree/src/test/obliterate.partialLength.spec.ts

```diff
@@ -175,211 +225,196 @@ describe("obliterate partial lengths", () => {
 				[
-					{ seq: refSeq, len: "hello world".length },
-					{ seq: refSeq + 1, len: "world".length, localSeq: refSeq + 1 },
-					{ seq: refSeq + 2, len: "world".length, localSeq: refSeq + 2 },
+					{ seq: refSeq, len: "hello world".length, localSeq: initialLocalSeq },
+					{ seq: refSeq, len: "world".length, localSeq: initialLocalSeq + 1 },
+					{ seq: refSeq + 1, len: "world".length, localSeq: initialLocalSeq },
+					{ seq: refSeq + 1, len: "world".length, localSeq: initialLocalSeq + 1 },
 				],
-				refSeq,
+				minRefSeqForLocalSeq,
 			);
-		});
 
-		it("passes for remote obliterate and local obliterate", () => {
-			obliterateRange({
-				mergeTree: client.mergeTree,
-				start: 0,
-				end: "hello ".length,
-				refSeq,
-				clientId: remoteClientId,
-				seq: refSeq + 1,
-				opArgs: undefined as never,
-			});
-			const localObliterateOp = client.obliterateRangeLocal(0, "hello".length);
+			client.applyMsg(client.makeOpMessage(localObliterateOp, refSeq + 2));
 
-			validatePartialLengths(localClientId, client.mergeTree, [
+			validatePartialLengths(NonCollabClient, client.mergeTree, [
 				{ seq: refSeq, len: "hello world".length },
 				{ seq: refSeq + 1, len: "world".length },
 				{ seq: refSeq + 2, len: "world".length },
 			]);
-
-			client.applyMsg(client.makeOpMessage(localObliterateOp, refSeq + 2));
-
-			validatePartialLengths(
-				remoteClientId,
-				client.mergeTree,
-				[
-					{ seq: refSeq, len: "hello world".length, localSeq: refSeq },
-					{ seq: refSeq + 1, len: "world".length, localSeq: refSeq + 1 },
-					{ seq: refSeq + 2, len: "".length, localSeq: refSeq + 2 },
-				],
-				refSeq,
-			);
+			validatePartialLengths(remoteClientId, client.mergeTree, [
+				{ seq: refSeq, len: "world".length },
+				{ seq: refSeq + 1, len: "world".length },
+				{ seq: refSeq + 2, len: "world".length },
+			]);
 		});
 	});
 
 	describe("obliterate with concurrent inserts", () => {
 		it("obliterates when concurrent insert in middle of string", () => {
 			const localObliterateOp = client.obliterateRangeLocal(0, client.getLength());
-			insertText({
-				mergeTree: client.mergeTree,
-				pos: "hello".length,
+
+			client.insertTextRemote(
+				"hello".length,
+				"more ",
+				undefined,
+				refSeq + 1,
 				refSeq,
-				clientId: remoteClientId,
-				seq: refSeq + 1,
-				text: "more ",
-				props: undefined,
-				opArgs: { op: { type: MergeTreeDeltaType.INSERT } },
-			});
+				client.getLongClientId(remoteClientId),
+			);
 			assert.equal(client.getText(), "");
 
-			validatePartialLengths(localClientId, client.mergeTree, [
-				{ seq: refSeq, len: "hello world".length },
-				{ seq: refSeq + 1, len: "hellomore  world".length },
-				{ seq: refSeq + 1, len: "".length, localSeq: refSeq + 1 },
+			const minRefSeqForLocalSeq = new Map<number, number>([
+				[initialLocalSeq, refSeq],
+				// TODO:AB#32300: We should be able to enable this or have a solid explanation on why it returns negative lengths.
```

Comment:
I think it's probably straightforward to fix this bug, but I didn't want to include any partial lengths changes that are not a direct result of merging obliterates + removes in this PR. Planning on fixing this right after, though.